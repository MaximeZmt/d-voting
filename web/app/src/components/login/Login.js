import {React, useContext, useState} from 'react';
import {Translations} from '../language/Translations';
import {LanguageContext} from '../language/LanguageContext';
import {SIGNIN_ENDPOINT} from '../utils/Endpoints';
import {GET_TEQ_EENDPOINT} from '../utils/ExpressEndoints';
import './Login.css';
import PropTypes from 'prop-types';


const Login = ({setToken}) => {
    const request = null;
    const [loginError, setLoginError] = useState();
    const [context, ] = useContext(LanguageContext);

    const handleClick = async() => {
        try{
            fetch(GET_TEQ_EENDPOINT).then(resp => {
                const json_data = resp.json();
                json_data.then(result => {
                    window.location = result['url'];
                });
            }).catch(error => {
                console.log(error);
            });
        } catch (error){
            console.log(error);
        }


        return (<div>
            {loginError === null? <div></div>: Translations[context].errorServerDown}
        </div>)
    }

    return (
        <div className='login-wrapper'>
            <div className='login-txt'>{Translations[context].loginText}</div>
            <button id='login-button' className='login-btn' onClick={handleClick}>{Translations[context].login}</button>
        </div>
    )
}

Login.propTypes = {
    setToken : PropTypes.func,
}

export default Login;