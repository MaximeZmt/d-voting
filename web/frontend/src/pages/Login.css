.login-container {
  min-width: 58ch;
  display: flex;
  text-align: center;
  flex-direction: column;
  padding-top: 2%;
  padding-left: 20%;
  padding-right: 20%;
  justify-content: center;
  background-color: whitesmoke;
  padding-bottom: 200px;
}
.login-txt {
  display: flex;
  padding-top: 5%;
  padding-bottom: 10%;
}
.login-btn {
  display: flex;
}
