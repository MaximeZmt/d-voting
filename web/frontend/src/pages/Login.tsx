import React, { FC, useState } from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import { GET_TEQ_EENDPOINT } from '../components/utils/ExpressEndoints';
import './Login.css';

const Login: FC = () => {
  const { t } = useTranslation();

  const [loginError] = useState();

  const handleClick = async () => {
    fetch(GET_TEQ_EENDPOINT)
      .then((resp) => {
        const jsonData = resp.json();
        jsonData.then((result) => {
          window.location = result.url;
        });
      })
      .catch((error) => {
        console.log(error);
      });

    return <div>{loginError === null ? <div></div> : t('errorServerDown')}</div>;
  };

  return (
    <div className="login-container">
      <div className="login-txt">{t('loginText')}</div>
      <button id="login-button" className="login-btn" onClick={handleClick}>
        {t('login')}
      </button>
    </div>
  );
};

Login.propTypes = {
  setToken: PropTypes.func,
};

export default Login;
