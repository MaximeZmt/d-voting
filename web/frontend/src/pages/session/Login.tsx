import React, { FC, useState } from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import { ENDPOINT_GET_TEQ_KEY } from 'components/utils/Endpoints';
import './Login.css';
import { useLocation } from 'react-router-dom';

const Login: FC = () => {
  const { t } = useTranslation();

  const [loginError] = useState();

  const [content, setContent] = useState('');

  // The backend will provide the client the URL to make a Tequila
  // authentication. We therefore redirect to this address.
  const handleClick = async () => {
    fetch(ENDPOINT_GET_TEQ_KEY)
      .then((resp) => {
        const jsonData = resp.json();
        jsonData.then((result) => {
          window.location = result.url;
        });
      })
      .catch((error) => {
        console.log(error);
      });

    return <div>{loginError === null ? <div></div> : t('errorServerDown')}</div>;
  };

  const location = useLocation();

  type LocationState = {
    from: Location;
  };

  useEffect(() => {
    const state = location.state as LocationState;
    if (state !== null) {
      setContent(t('loginText', { from: state.from.pathname }));
    }
  }, [location]);

  return (
    <div className="login-container">
      <div className="login-txt">{content}</div>
      <button id="login-button" className="login-btn" onClick={handleClick}>
        {t('login')}
      </button>
    </div>
  );
};

Login.propTypes = {
  setToken: PropTypes.func,
};

export default Login;
