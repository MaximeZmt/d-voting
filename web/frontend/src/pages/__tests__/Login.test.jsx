import React from 'react';
import Login from '../Login';
import renderer from 'react-test-renderer';

describe('Login', () => {
  it('should render the Login Component correctly', () => {
    const component = renderer.create(<Login />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
