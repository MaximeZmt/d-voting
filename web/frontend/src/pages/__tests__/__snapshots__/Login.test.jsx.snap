// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the Login Component correctly 1`] = `
<div
  className="login-wrapper"
>
  <div
    className="login-txt"
  >
    loginText
  </div>
  <button
    className="login-btn"
    id="login-button"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
